tests/plugins/neg/divideZero/plugin_1.scala

```diff
@@ -0,0 +1,37 @@
+import dotty.tools.dotc._
+import core._
+import Contexts.Context
+import plugins.Plugin
+import Phases.Phase
+import ast.tpd
+import transform.MegaPhase.MiniPhase
+import Decorators._
+import Symbols.Symbol
+import Constants.Constant
+
+class DivideZero extends MiniPhase with Plugin {
+  val name: String = "divideZero"
+  override val description: String = "divide zero check"
+
+  val phaseName = name
+
+  override def init(phases: List[List[Phase]])(implicit ctx: Context): List[List[Phase]] = {
+    val (before, after) = phases.span(ps => !ps.exists(_.phaseName == "pickler"))
```

Comment:
It would be better to not rely on strings to identify phase names and use class types instead, similar to what we do with `Phase.replace` in https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/repl/ReplCompiler.scala#L40

Reply:
Strings compose with phases coming from other compiler plugins. Class types do not.

Reply:
@sjrd Good point. But can't we use `Class.forName` in that case?

Reply:
If all compiler plugins are loaded in the same `ClassLoader`, then yes. But is that the case and is it desirable?